File: next/courses/types/Course.ts
export interface Course {
    id: string;
    title: string;
    description: string;
    createdAt: string;
    imageUrl: string;
    u: {
        id: string;
        name: string;
    };
    tags: string[];
    isPublished?: boolean;
    versionId: string;
    versionName: string;
}

File: kotlin/coretypings/src/main/kotlin/eduflow/course/course/Course.kt
package flowledge.course.course

import flowledge.course.CourseCreator
import java.util.*

interface Course {
    val _id: String
    val title: String
    val description: String
    val imageUrl: String?
    val u: CourseCreator
    val createdAt: Date
    val updatedAt: Date
    val tags: List<String>?
    val isPublished: Boolean?
    val versions: List<String>
}